receiver/otelarrowreceiver/metadata.yaml

```diff
@@ -8,3 +8,29 @@ status:
   distributions: [contrib]
   codeowners:
     active: [jmacd, moh-osman3]
+
+telemetry:
+  metrics:
+    otel_arrow_receiver_in_flight_bytes:
+      description: Number of bytes in flight
+      unit: By
+      enabled: true
+      sum:
+        monotonic: false
+        value_type: int
+
+    otel_arrow_receiver_in_flight_items:
+      description: Number of items in flight
+      enabled: true
+      unit: 1
```

Comment:
Would it make sense for unit to be `{item}` here? Same thought below to use a unit of `{request}` for number of requests in flight.

(It could be done in a follow up if necessary, since these are existing metrics.)